packages/plugin/src/theia.d.ts

```diff
@@ -6209,6 +6209,19 @@ declare module '@theia/plugin' {
         * @return A [disposable](#Disposable) that unregisters this provider when being disposed.
         */
         export function registerRenameProvider(selector: DocumentSelector, provider: RenameProvider): Disposable;
+
+        /**
+         * Register a call hierarchy provider.
+         *
+         * Multiple provider can be registered for a language. In that case providers are asked in
+         * parallel and the results are merged. A failing provider (rejected promise or exception) will
+         * not cause a failure of the whole operation.
+         *
+         * @param selector A selector that defines the documents this provider is applicable to.
+         * @param service A call hierarchy provider.
+         * @return A [disposable](#Disposable) that unregisters this provider when being disposed.
+         */
+        export function registerCallHierarchyProvider(selector: DocumentSelector, provider: CallHierarchyProvider): Disposable;
```

Comment:
Is it proposed api? Please move it theia.proposed.d.ts

Reply:
We probably should copy it from here: https://github.com/microsoft/vscode-languageserver-node/blob/51ce5233b6f97a4e2b6bee8f729205774a243140/client/src/callHierarchy.proposed.ts#L185-L234

Please don't forget a CQ. ping @theia-ide/ip-help if you need any help with it